### PR TITLE
Introduce `info` and `warning` common utilities

### PR DESCRIPTION
**Pull request by literat**
## Description

Introduce `info` and `warning` common utilities and refactor the usage of the `Console` functions

### Additional context

- https://www.npmjs.com/package/tiny-warning
- https://github.com/zertosh/invariant/tree/master
- https://github.com/alexreardon/tiny-warning/tree/master

### Issue reference

https://jira.lmc.cz/browse/DS-767

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/common/utilities/warning.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  if (!isProduction()) {
+    // condition passed: do not log
+    if (condition) {
+      return;
+    }
+
+    // Condition not passed
+    const text = `Warning: ${message}`;
+
+    // check console for IE9 support which provides console
```
Comment (adamkudrna) [inline packages/common/utilities/warning.ts:15]: IE9? 🙂

File: packages/common/utilities/warning.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    // only with open devtools
```
Comment (adamkudrna) [inline packages/common/utilities/warning.ts:16]: Why just when dev tools are open? Isn't a warning useful at all times during development…?

File: packages/common/utilities/warning.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    if (typeof console !== 'undefined') {
+      // we want to log a warning; so usage of Console is required
+      // eslint-disable-next-line no-console
+      console.warn(text);
+    }
+
+    // Throwing an error and catching it immediately
+    // to improve debugging
+    // A consumer can use 'pause on caught exceptions'
```
Comment (adamkudrna) [inline packages/common/utilities/warning.ts:25]: 👍🏻

File: packages/common/utilities/info.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    // Condition not passed
+    const text = `Info: ${message}`;
+
+    // check console for IE9 support which provides console
+    // only with open devtools
+    if (typeof console !== 'undefined') {
+      // we want to log a warning; so usage of Console is required
+      // eslint-disable-next-line no-console
```
Comment (adamkudrna) [inline packages/common/utilities/info.ts:19]: I believe it's better to stick the reason to the disabling comment:

```suggestion
      // eslint-disable-next-line no-console -- We want to log a warning; so usage of Console is required.
```

Copy-pasting will be safer and we can do fulltext search on all disabling comments that do not have a reason next to it.

https://eslint.org/docs/latest/use/configure/rules#comment-descriptions

File: packages/common/utilities/__tests__/warning.test.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+// we are testing Console function
+/* eslint-disable no-console */
```
Comment (adamkudrna) [inline packages/common/utilities/__tests__/warning.test.ts:2]: Again, reasons should follow the disabling statement so they are better connected to it.

https://eslint.org/docs/latest/use/configure/rules#comment-descriptions

**Review by adamkudrna — APPROVED** [review]
I'd only wish to improve the disabling comments so they have the reason right after them. Otherwise looks good to me. 👍🏻

File: packages/common/constants/environments.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export const ENVIRONMENTS = {
+  DEVELOPMENT: 'development',
+  TESTING: 'testing',
+  PRODUCTION: 'production',
+};
+
+export const isTesting = () => process.env.NODE_ENV === ENVIRONMENTS.TESTING;
+export const isProduction = () => process.env.NODE_ENV === ENVIRONMENTS.PRODUCTION;
```
Comment (crishpeen) [inline packages/common/constants/environments.ts:8]: please add isDevelopment too

Comment (literat) [reply packages/common/constants/environments.ts:9]: Ok, added.

File: packages/common/utilities/warning.ts (hunk 1 above)
Comment (literat) [reply packages/common/utilities/warning.ts]: 🤷

File: packages/common/utilities/info.ts (hunk 4 above)
Comment (literat) [reply packages/common/utilities/info.ts]: Fixed.

File: packages/common/utilities/__tests__/warning.test.ts (hunk 5 above)
Comment (literat) [reply packages/common/utilities/__tests__/warning.test.ts]: Fixed.

**Pull request merged**